### ISSUE TITLE
Handling transitive dependencies from other glided projects is confusing

### ISSUE THREAD

**Issue by JacekLach** [Masterminds/glide#50]
I can't figure out how to properly add a project that is managed by glide as a dependency.

Say for example I have a local library A that is managed by glide, and in it's glide.yaml it specifies a dependency on B.

If `glide in` followed by `go build` in A everything works as expected.

Now, if in another project I add A as a dependency (to glide.yaml), and do `glide install`, it will fail because B isn't present (`cannot find package "<B>" in any of: $GOROOT $GOPATH`).

It appears the flow to make this kind of work is:

```
glide in
glide install # ignore 'failed to run go install' errors
go get # fetch transitive dependencies
go build # rebuild dependencies
```

Now, that only works if `go get` can fetch the transitive dependency B; i.e. it's a simple `package: <B>` entry in glide.yaml. I don't see a way of making this work if B is a more complex dependency (i.e. specifies repository/branch/vcs).

Am I missing something obvious? Are you only supposed to use glide for the top-level application, not for any supporting libraries?

**Comment by mattfarina** [comment]
Right now `glide.yaml` is only for the top level application. We don't iterate over the dependent projects.

I would like to add this functionality. I'm shooting for the 0.5 release but there might be enough other changes in that release for it to wait for the 0.6 release. Lots of good/fun/useful things are coming.

**Comment by technosophos** [comment]
Have you tried `glide guess` for resolving dependencies?

**Comment by technosophos** [comment]
Now that 0.5 is out, I'll do my best to answer. Up until recently, the approach that we've used is that if Project B has other dependencies, the top-level project needs to have them in its glide.yaml file.

To make it slightly easier, we add `glide guess`, which will attempt to figure out all of those dependencies for you, and emit glide.yaml data.

In the newest versions, we added `glide get --import`, which will attempt to import Godeps and GPM dependency files, too. But we're still working on an even better way to fetch deeply nested dependencies.

If you have an idea for how you think it ought to work (and I think you sorta hinted at it by calling these dependency types "transitive"), let us know. We have been encouraging a more explicit treatment of dependencies, and hadn't really focused on non-reproducible builds.

**Comment by technosophos** [comment]
And we're now working on `glide guess --get`, which will do a deep analysis of the code, fetching transitive dependencies as it goes.

**Comment by technosophos** [comment]
With the new `feature/flatten` branch, this will now automatically introspect code and pull down transitive dependencies. As soon as that is merged, we can mark this issue closed.

**Comment by technosophos** [comment]
I believe this is fixed completely in Glide >= 0.9.0

**Comment by mattfarina** [comment]
I think this is fixed. If not please re-open.
